pkg/istio/onboarding/controller.go

```diff
@@ -181,14 +180,22 @@ func (c *Controller) getOnboardedServiceList() []string {
 	return serviceList
 }
 
-// errHandler re-adds the key for a failed processor.sync operation
-func (c *Controller) errHandler(err error, item *processor.Item) error {
-	if err != nil {
-		if item != nil {
-			log.Errorf("%s Error performing %s on %s: %s", logPrefix, item.Operation, item.Resource.Key(), err)
-		}
-		c.queue.AddRateLimited(queueKey)
+// callbackHandler re-adds the key for a failed processor.sync operation
+func (c *Controller) callbackHandler(err error, item *processor.Item) error {
+	if err == nil {
+		return nil
+	}
+	if item != nil {
+		log.Errorf("%s Error performing %s on %s: %s", logPrefix, item.Operation, item.Resource.Key(), err)
+	}
+	if c.queue.NumRequeues(queueKey) >= queueNumRetries {
+		log.Errorf("%s Max number of retries reached for %s.", logPrefix, queueKey)
```

Comment:
I think we need alerts for this case since there is only one key when it comes to this controller.